### ISSUE TITLE
Instead of a singleton observer bloc, use an observer using zones

### ISSUE THREAD

**Issue by PlugFox** [PlugFox/bloc#2]
Instead of a singleton observer bloc, use an observer using zones.

Like this:
```dart
Bloc.observe(() {
  runApp(App());
});
```

**Comment by Aqluse** [comment]
Can we please give consideration to pros and cons of both approaches more accurately and put it down into this issue?

**Comment by PlugFox** [comment]
@Aqluse 
BlocObserver as is - just setting into mutable singleton static field.

```
Bloc.observer = MyObserver();
runApp(App());
```

This is not very nice behavior.

But we can use Zones.
Find the nearest observer through them and use it.

```dart
Bloc.observe(() {
  runApp(App());
});
```

**Comment by PlugFox** [comment]
Implemented in #1
